service/src/test/java/uk/nhs/adaptors/gp2gp/ehr/mapper/diagnosticreport/DiagnosticReportMapperTest.java

```diff
@@ -341,7 +377,23 @@ private Answer<String> mockIdForReference() {
     private Answer<String> mockSpecimenMapping() {
         return invocation -> {
             Specimen specimen = invocation.getArgument(0);
-            return String.format("<!-- Mapped Specimen with id: %s -->", specimen.getId());
+            List<Observation> observations = invocation.getArgument(1);
+
+            List<String> linkedObservations = new ArrayList<>();
+
+            for (Observation observation : observations) {
+                if (observation.getSpecimen().getReference() != null
+                        && observation.getSpecimen().getReference().equals(specimen.getId())) {
+                    linkedObservations.add(observation.getId());
+                }
+            }
+
+            if (linkedObservations.isEmpty()) {
+                return String.format("<!-- Mapped Specimen with id: %s -->", specimen.getId());
+            }
```

Comment:
Does this code ever get used? Surely in all our cases now we'll be passing in a linked Observation?

Reply:
Alex: All Observations will have a Specimen after this change, but not all Observations will have a Specimen.